addons/mass_mailing/models/mailing_trace.py

```diff
@@ -96,9 +96,11 @@ class MailingTrace(models.Model):
         ("mail_smtp", "Connection failed (outgoing mail server problem)"),
         # mass mode
         ("mail_bl", "Blacklisted Address"),
-        ("mail_optout", "Opted Out"),
+        ("mail_bounce", "Bounce"),
```

Comment:
Same here, we have a trace_status bounce and a failure_type bounce.

Reply:
Well it is not unknown ... currently under a big umbrella "bounce" but for sure we could improve our heuristic and add some more details (out of storage, ...).